lib/run-status.js

```diff
@@ -99,10 +105,16 @@ class RunStatus extends Emittery {
 
 				stats.remainingTests--;
 				fileStats.remainingTests--;
+				this.pendingTests.get(event.testFile).delete(event.title);
 				break;
 			case 'timeout':
+				event.tests = this.pendingTests;
+				this.pendingTests = new Map();
```

Comment:
I'm not at all opposed to tracking this stuff here, pending a larger overhaul of the reporters. However there may be race conditions where we time out one worker, and receive events from another right before we make it exit. I don't think you can reset `pendingTests` like this without adding guards in the other case statements, and that starts to feel iffy.

That's probably why the logic was in the (verbose) reporters.

`event.tests` is perhaps not the best naming either. Perhaps `activeTests` or `concurrentTests`? But then again it may be best to revert to the existing logic. We can backport to the mini reporter later.

Reply:
I'll look into it and revert back if I can't find a way to work around the race condition you mentioned. Wanted to avoid duplicating the code in the reporters.

Reply:
I've fixed the naming and added some checks. The reset shouldn't create problems. It's basically the same implementation as before, except its in RunStatus instead of the reporter. I think that it's more fitting for it to be in RunStatus, and it spares us of some code duplication.

Take a look at the diffs and if you still want it reverted, it's no big deal either way.